src/lib.rs

```diff
@@ -75,7 +85,7 @@ impl OfferHandler {
         blinded_path: BlindedPath,
         reply_path: Option<BlindedPath>,
     ) -> Result<(), OfferError<Secp256k1Error>> {
-        sleep(Duration::from_secs(5)).await;
+        self.wait_for_ready().await;
```

Comment:
meta-nit: would be useful to have a commit message here to add some context to what we're doing